Blammo-wai/src/Network/Wai/Middleware/Logging.hs

```diff
@@ -144,10 +144,39 @@ requestLoggerWith config env app req respond =
     app req $ \resp -> do
       recvd <- respond resp
       duration <- toMillis . subtract begin <$> getTime
-      recvd <$ runInIO (runWithLogger env $ logResponse config duration req resp)
+      runInIO $
+        runWithLogger env $
+          if isRaw resp
+            then logRawResponse config duration req
+            else logResponse config duration req resp
+      pure recvd
  where
   getTime = Clock.getTime Clock.Monotonic
   toMillis x = fromIntegral (Clock.toNanoSecs x) / nsPerMs
+  isRaw = \case
+    ResponseRaw {} -> True
+    _ -> False
```

Comment:
Might be a good idea to list all the constructors separately instead of using a wildcard, especially since this is imported from an "internal" module, to force reconsidering if the constructors change.

Reply:
Done